### PR TITLE
Feature dataviewer window listener

### PR DESCRIPTION
**Pull request by RalphSteinhagen**
new features: 

* explicit listener to DataViewWindow state changes

* added safe-guards and some bug-fixes

### CODE REVIEW

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/chart-fx/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/chart-fx/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/chart-fx/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/chart-fx/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/chart-fx/chart-fx/pullRequest?prid=4592117&bid=15418510)

File: chartfx-chart/src/main/java/de/gsi/chart/XYChart.java [hunk 1]
```diff
@@ -5,14 +5,15 @@
 import java.security.InvalidParameterException;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Objects;
+import java.util.Optional;
 import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
```
Comment (RalphSteinhagen) [inline chartfx-chart/src/main/java/de/gsi/chart/XYChart.java]: Codacy Issue found: [Unused import - java.util.stream.Stream.](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/chart-fx/chart-fx/pullRequest?prid=4592117&bid=15418510)

File: chartfx-chart/src/main/java/de/gsi/chart/XYChart.java (hunk 1 above)
Comment (RalphSteinhagen) [inline chartfx-chart/src/main/java/de/gsi/chart/XYChart.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Stream'](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/chart-fx/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/chart-fx/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/chart-fx/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/renderer/datareduction/MaxDataReducer.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/renderer/datareduction/MaxDataReducer.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/renderer/datareduction/MaxDataReducer.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117&bid=15418510)

File: chartfx-chart/src/main/java/de/gsi/chart/viewer/DataView.java [hunk 2]
```diff
@@ -72,21 +78,31 @@ public void set(final DataViewWindow newNode) {
         }
     };
 
-    public DataView(final String name, final Node icon) {
+    @Deprecated // is this default constructor really needed?
```
Comment (wirew0rm) [inline chartfx-chart/src/main/java/de/gsi/chart/viewer/DataView.java]: Needed only if we want to allow only giving name or icon. I added this, when I was not yet fully aware of the intended use of the DataView, i think we should remove them before merge to keep the api clean.

Comment (RalphSteinhagen) [reply chartfx-chart/src/main/java/de/gsi/chart/viewer/DataView.java]: incorporated with commit 05ba35673e7f8fd98fb620daf0dc086d874e5584

**Comment by RalphSteinhagen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoredEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMaximisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowRestoringEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/DataViewWindow.java  2
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowClosedEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowDetachingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowUpdateEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/viewer/event/WindowMinimisingEvent.java  1
- chartfx-chart/src/main/java/de/gsi/chart/renderer/datareduction/MaxDataReducer.java  1
         

Complexity decreasing per file
==============================
+ chartfx-chart/src/main/java/de/gsi/chart/Chart.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4592117&bid=15418510)

**Pull request merged**